### PR TITLE
Add EyeLike Ivy Functional API

### PR DESCRIPTION
**Pull request by a-elrawy**
close https://github.com/unifyai/ivy/issues/9976

### CODE REVIEW

**Comment by a-elrawy** [comment]
Closes #9976
Tracking TODO issue #8936

File: ivy/functional/ivy/creation.py [hunk 1]
```diff
@@ -1100,6 +1100,158 @@ def eye(
...
+               [0., 1.]])
+
+    >>> x1 = ivy.array([0, 1, 2],[3, 4, 5],[6, 7, 8])
+    >>> y1 = ivy.eye_like(x1)
+    >>> print(y1)
+    ivy.array([[1., 0., 0.],
+               [0., 1., 0.],
+               [0., 0., 1.]])
+
+    With :'k' input:
```
Comment (hirwa-nshuti) [inline ivy/functional/ivy/creation.py:1172]: You shouldn't mention `k` as input example alone you can only use `ivy` instances means you can either mix it with `ivy.Array` examples

File: ivy/functional/ivy/creation.py [hunk 2]
```diff
@@ -1100,6 +1100,158 @@ def eye(
...
+
+
+    >>> x2 = ivy.native_array([3, 8],[0, 2], [8, 5]])
+    >>> y2 = ivy.eye_like(x2, dtype=ivy.IntDtype('int32'), device=ivy.Device('cpu'))
+    >>> print(y2)
+    ivy.array([[1., 0.],
+               [0., 1.],
+               [0., 0.]])
+
+    # Array ``y2`` is now stored on the CPU.
```
Comment (hirwa-nshuti) [inline ivy/functional/ivy/creation.py:1207]: These comments should be removed as well

File: ivy/functional/ivy/creation.py [hunk 3]
```diff
@@ -1100,6 +1100,158 @@ def eye(
...
+    >>> y = ivy.eye_like(x)
+    >>> print(y)
+    {
+        a: ivy.array([[1., 0.],
+                      [0., 1.]]),
+        b: ivy.array([[1., 0.],
+                      [0., 1.]])
+    }
+
+    Instance Method Examples
```
Comment (hirwa-nshuti) [inline ivy/functional/ivy/creation.py:1221]: Instance method examples should be put on the methods docstrings.

File: ivy/functional/ivy/creation.py (hunk 1 above)
Comment (a-elrawy) [reply ivy/functional/ivy/creation.py:1172]: okay, thanks a lot for the clarification

File: ivy/functional/ivy/creation.py [hunk 4]
```diff
@@ -1100,6 +1100,158 @@ def eye(
...
+    With :'k' input:
+
+    >>> x1 = ivy.array([0, 1, 2],[3, 4, 5],[6, 7, 8])
+    >>> y1 = ivy.eye_like(x1, k=1)
+    >>> print(y1)
+    ivy.array([[0., 1., 0.],
+               [0., 0., 1.],
+               [0., 0., 0.]])
+
+    With :'out' input:
```
[the remark this answers is not in the document]
Comment (a-elrawy) [reply ivy/functional/ivy/creation.py:1181]: okay, I removed it

File: ivy/functional/ivy/creation.py (hunk 2 above)
Comment (a-elrawy) [reply ivy/functional/ivy/creation.py:1207]: done

File: ivy/functional/ivy/creation.py (hunk 3 above)
Comment (a-elrawy) [reply ivy/functional/ivy/creation.py:1221]: Can you please elaborate more? I was following the code in zero_like as a reference

Comment (a-elrawy) [reply ivy/functional/ivy/creation.py]: Thanks for explanation.

**Comment by a-elrawy** [comment]
Hi @hirwa-nshuti, After moving to Experimental API, the tests are failing because It doesn't recognize the function any more. How Can I solve this issue?

PS: I think the reason is that I implemented it as compositional function and It doesn't have backend-specific implementation, but I'm not sure about this

**Comment by hirwa-nshuti** [comment]
> Hi @hirwa-nshuti, After moving to Experimental API, the tests are failing because It doesn't recognize the function any more. How Can I solve this issue?
> 
> PS: I think the reason is that I implemented it as compositional function and It doesn't have backend-specific implementation, but I'm not sure about this

Hey it is a problem with `number_positional_args` in the test function you can specify it under `@handle_test`
just use `number_positional_args=st.just(1)` the tests will pass

**Comment by a-elrawy** [comment]
> > Hi @hirwa-nshuti, After moving to Experimental API, the tests are failing because It doesn't recognize the function any more. How Can I solve this issue?
> > PS: I think the reason is that I implemented it as compositional function and It doesn't have backend-specific implementation, but I'm not sure about this
> 
> Hey it is a problem with `number_positional_args` in the test function you can specify it under `@handle_test` just use `number_positional_args=st.just(1)` the tests will pass

it worked. Thanks a lot for your help

File: ivy/functional/ivy/experimental/creation.py [hunk 5]
```diff
@@ -475,3 +476,97 @@ def tril_indices(
...
+        optional output array, for writing the result to. It must have a shape that the
+        inputs broadcast to.
+
+    Returns
+    -------
+    ret
+        an array having the same shape as ``x`` and filled with ``ones`` in diagonal ``k``
+        and ``zeros`` elsewhere.
+
+    This function conforms to the `Array API Standard
```
Comment (hirwa-nshuti) [inline ivy/functional/ivy/experimental/creation.py:523]: This function doesn't conform to Array API. Feel free to refer to onnx docs and update docstrings.
https://github.com/onnx/onnx/blob/main/docs/Operators.md#EyeLike

After updating docstrings the changes will be good to be merged🙂

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Check a few changes needed from comments😊

File: ivy/functional/ivy/experimental/creation.py (hunk 5 above)
Comment (a-elrawy) [reply ivy/functional/ivy/experimental/creation.py]: sorry, I didn't notice it. I changed it now, Should I add also onnx docs link or the description is enough?

Comment (hirwa-nshuti) [reply ivy/functional/ivy/experimental/creation.py]: Description is enough

File: ivy/functional/ivy/experimental/creation.py [hunk 6]
```diff
@@ -475,3 +476,102 @@ def tril_indices(
...
+    Generate a 2D tensor (matrix) with ones on the diagonal and zeros everywhere else.
+    Only 2D tensors are supported, i.e. input T1 must be of rank 2. The shape of the
+    output tensor is the same as the input tensor. The data type can be specified by
+    the 'dtype' argument. If 'dtype' is not specified, then the type of input tensor
+    is used. By default, the main diagonal is populated with ones, but attribute 'k'
+    can be used to populate upper or lower diagonals.
+
+
+    Both the description and the type hints above assumes an array input for simplicity,
+    but this func;tion is *nestable*, and therefore also accepts :class:`ivy.Container`
```
Comment (hirwa-nshuti) [inline ivy/functional/ivy/experimental/creation.py:534]:
```suggestion
    but this function is *nestable*, and therefore also accepts :class:`ivy.Container`
```

**Review by hirwa-nshuti — APPROVED** [review]
Looks good to me😊
Thanks for contributing🙂

**Pull request merged**